src/dda/utils/platform/_pty/unix.py

```diff
@@ -39,6 +41,10 @@ def __init__(
         self._fd, child_fd = pty.openpty()
         os.set_inheritable(self._fd, False)
         os.set_inheritable(child_fd, True)
+
+        width, height = terminal_size()
+        termios.tcsetwinsize(self._fd, (height, width))
```

Comment:
Luckily this was introduced recently in 3.11. If this didn't exist we could do something like the following (which I was about to do until I discovered it):

```python
s = struct.pack('HHHH', height, width, 0, 0)
fcntl.ioctl(fd, termios.TIOCSWINSZ, s)
```